packages/vscode-extension/src/common/License.ts

```diff
@@ -1,16 +1,70 @@
 // #region License Helpers
+export enum FeatureAvailabilityStatus {
+  Available,
+  InsufficientLicense,
+}
+
+export function getFeatureAvailabilityStatus(
+  licenseStatus: LicenseStatus,
+  feature: Feature
+): FeatureAvailabilityStatus {
+  switch (licenseStatus) {
+    case LicenseStatus.Inactive:
+    case LicenseStatus.Free:
+      return FreeFeatures.has(feature)
+        ? FeatureAvailabilityStatus.Available
+        : FeatureAvailabilityStatus.InsufficientLicense;
+    case LicenseStatus.Pro:
+      return ProFeatures.union(FreeFeatures).has(feature)
```

Comment:
I know we're in JS land, and performance is a pipe dream, but creating a set union each time this check is performed (and doing it _twice_ in the `Team` branch and _three times_ in the `Enterprise`) sounds really excessive.
I'm also not 100% sure how I feel about splitting the features between maps -- in my mind, it'd make more sense for `ProFeatures` to simply include all of the `FreeFeatures` as well. I guess it might be useful to check if a feature is `Pro` but not `Free`, but I also feel like that's the case that can take a slower path (of checking both maps and returning `Pro.has && !Free.has`), rather than this one.